src/main/scala/spinoco/fs2/mail/internal/internal.scala

```diff
@@ -0,0 +1,43 @@
+package spinoco.fs2.mail
+
+import fs2.{Pipe, Stream}
+import fs2.util.Async
+import fs2.util.syntax._
+
+package object internal {
+
+  /**
+    * Behaves like [[Stream.takeThrough]], but even if down stream ends earlier than the `toggle`
+    * results into false. This will ensure that the [[Stream]] will not end before we get false from the `toggle`.
```

Comment:
resulting stream will not end untils `f` yields to false, draining any elements where `f` still yeilds to ....